backend/app/usecase/search/search.go

```diff
@@ -66,13 +69,108 @@ func (s Search) searchResource(resource Resource, orderBy order.Order, query Que
 }
 
 func (s Search) searchShortLink(query Query, orderBy order.Order, filter Filter) (Result, error) {
-	return Result{}, nil
+	if query.User == nil {
+		return Result{}, errors.New("user not provided")
+	}
+	if len(query.Query) == 0 {
+		return Result{}, errors.New("query not provided")
```

Comment:
This may be not considered as an error, if a user sends an empty query then all the short links will be returned. When I was developing it in the first place, I had in mind different logic for matcher

Reply:
@arberiii This sound good for now.

Reply:
We may apply different strategy in the future.